src/m365/spo/commands/applicationcustomizer/applicationcustomizer-get.ts

```diff
@@ -0,0 +1,177 @@
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.id && !validation.isValidGuid(args.options.id)) {
+          return `${args.options.id} is not a valid GUID`;
+        }
+
+        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['title', 'id', 'clientSideComponentId'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const customAction = await this.getCustomAction(args.options);
+
+      if (customAction) {
+        logger.log({
+          ClientSideComponentId: customAction.ClientSideComponentId,
+          ClientSideComponentProperties: customAction.ClientSideComponentProperties,
+          CommandUIExtension: customAction.CommandUIExtension,
+          Description: customAction.Description,
+          Group: customAction.Group,
+          Id: customAction.Id,
+          ImageUrl: customAction.ImageUrl,
+          Location: customAction.Location,
+          Name: customAction.Name,
+          RegistrationId: customAction.RegistrationId,
+          RegistrationType: customAction.RegistrationType,
+          Rights: JSON.stringify(customAction.Rights),
+          Scope: this.humanizeScope(customAction.Scope),
+          ScriptBlock: customAction.ScriptBlock,
+          ScriptSrc: customAction.ScriptSrc,
+          Sequence: customAction.Sequence,
+          Title: customAction.Title,
+          Url: customAction.Url,
+          VersionOfUserCustomAction: customAction.VersionOfUserCustomAction
+        });
+      }
+    }
+    catch (err: any) {
+      this.handleRejectedPromise(err);
+    }
+  }
+
+  private async getCustomAction(options: Options): Promise<CustomAction | undefined> {
+    const scope = options.scope ? options.scope : 'All';
+
+    if (options.id) {
+      const customAction = await spo.getCustomActionById(options.webUrl, options.id, scope);
+
+      if (!customAction) {
+        throw `No application customizer with id '${options.id}' found`;
+      }
+
+      if (customAction.Location !== 'ClientSideExtension.ApplicationCustomizer') {
+        throw 'The found custom action is not an Application Customizer';
+      }
+
+      return customAction;
+    }
+
+    const filter = options.title ? `Title eq '${formatting.encodeQueryParameter(options.title as string)}'` : `ClientSideComponentId eq guid'${formatting.encodeQueryParameter(options.clientSideComponentId as string)}'`;
+    const customActions = await spo.getCustomActions(options.webUrl, scope, filter);
+
+    if (customActions.length === 1) {
+      if (customActions[0].Location !== 'ClientSideExtension.ApplicationCustomizer') {
+        throw 'The found custom action is not an Application Customizer';
+      }
```

Comment:
same here I would also include this as an additional filter passed to `spo.getCustomAcions` method like `Location eq 'Location eq 'ClientSideExtension.ApplicationCustomizer` so we will consider only those